tests/nnstreamer_edge/nnstreamer-edge-custom-test.c

```diff
@@ -0,0 +1,229 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/**
+ * Copyright (C) 2024 Gichan Jang <gichan2.jang@samsung.com>
+ *
+ * @file   nnstreamer-edge-custom-test.c
+ * @date   30 Aug 2024
+ * @brief  NNStreamer-edge custom connection for test.
+ * @see    https://github.com/nnstreamer/nnstreamer
+ * @author Gichan Jang <gichan2.jang@samsung.com>
+ * @bug    No known bugs except for NYI items
+ */
+
+#include "nnstreamer-edge.h"
+#include "nnstreamer-edge-custom.h"
+#include "nnstreamer_log.h"
+#include "nnstreamer_util.h"
+#include <glib.h>
+
+#define SAFE_FREE(p) do { if (p) { free (p); (p) = NULL; } } while (0)
+
+typedef struct
+{
+  int is_connected;
+  char *peer_address;
+  nns_edge_event_cb event_cb;
+  void *user_data;
+} nns_edge_custom_test_s;
+
+static int
+nns_edge_custom_close (void *priv)
+{
+  nns_edge_custom_test_s *custom_h;
+  if (!priv) {
+    nns_loge ("Invalid param, handle should not be null.");
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+  custom_h = (nns_edge_custom_test_s *) priv;
+
+  SAFE_FREE (custom_h->peer_address);
+  SAFE_FREE (custom_h);
+
+  return NNS_EDGE_ERROR_NONE;
+}
+
+static const char *
+nns_edge_custom_get_description (void)
+{
+  return "custom";
+}
+
+static int
+nns_edge_custom_create (void **priv)
+{
+  nns_edge_custom_test_s *custom_h;
+  if (!priv) {
+    nns_loge ("Invalid param, handle should not be null.");
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+
+  custom_h = (nns_edge_custom_test_s *) calloc (1, sizeof (nns_edge_custom_test_s));
+  if (!custom_h) {
+    nns_loge ("Failed to allocate memory for edge custom handle.");
+    return NNS_EDGE_ERROR_OUT_OF_MEMORY;
+  }
+
+  *priv = custom_h;
+
+  return NNS_EDGE_ERROR_NONE;
+}
+
+static int
+nns_edge_custom_start (void *priv)
+{
+  nns_edge_custom_test_s *custom_h;
+  if (!priv) {
+    nns_loge ("Invalid param, handle should not be null.");
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+  custom_h = (nns_edge_custom_test_s *) priv;
+  custom_h->is_connected = 0;
```

Comment:
This is just my personal opinion. :smile: 
How about changing the variable from ```is_connected``` to ```connect_status```and assigning the enum value?
{START, STOP, CONNECT}

Reply:
It's a good suggestion, but it's not necessary because it's just a test app.